### PR TITLE
Chore/ledgertime

### PR DESCRIPTION
**Pull request by allanbrondumkr**
* Adjusted ledger close times to be of type `u32` and defined type alias `LedgerTimestamp` for it
* Moved `owner_funds` from `TransactionCommon` to `CreateOfferTransaction`, see documentation after <https://xrpl.org/subscribe.html#order-book-streams>

It was not 100% clear what the ledger time data type is. But on offer types it is clear it is `u32` <https://xrpl.org/offercreate.html> and <https://xrpl.org/offer.html#offer-fields>. Also the rippled implementation uses 32 bits: <https://github.com/XRPLF/rippled/blob/develop/src/ripple/basics/chrono.h#L53>, <https://github.com/XRPLF/rippled/blob/develop/src/ripple/ledger/ReadView.h#L114>. The value `u32::MAX` represents the date time 2136-02-07 06:28:15 UTC

### CODE REVIEW

File: xrpl_types/src/lib.rs [hunk 1]
```diff
@@ -1,16 +1,18 @@
 //! Core types and related functions for the XRP Ledger. Reused between Web and
 //! WebSocket clients and potentially for server-side code.
 
-pub mod account;
-pub mod amount;
-pub mod book;
-pub mod currency;
-pub mod ledger_index;
-pub mod transaction;
+mod account;
```
Comment (allanbrondumkr) [inline xrpl_types/src/lib.rs:4]: I think the convention is not to have public modules if you "`pub use`" their contents. Else the types will be represented at multiple paths.

File: xrpl_api/src/lib.rs [hunk 2]
```diff
@@ -1,8 +1,8 @@
 //! A low-level crate that extracts the common RPC protocol used by the HTML
 //! and WebSocket clients.
 
-pub mod api;
-pub mod types;
+mod api;
+mod types;
 
 pub use api::*;
```
Comment (gmosx) [inline xrpl_api/src/lib.rs:7]: I am wondering if we could remove the `pub use` instead, thoughts?

File: xrpl_types/src/lib.rs (hunk 1 above)
Comment (gmosx) [reply xrpl_types/src/lib.rs:4]: I wonder if it would be better to remove the `pub use` instead?

File: xrpl_api/src/lib.rs (hunk 2 above)
Comment (allanbrondumkr) [reply xrpl_api/src/lib.rs:7]: I think the `pub use` is good in many cases. It means that you don't expose the internal module structure to api consumers. That is generally good.

Comment (allanbrondumkr) [reply xrpl_api/src/lib.rs:7]: I like it in the `xrpl_api` and `xrpl_types` crates, do you see any downsides?

Comment (gmosx) [reply xrpl_api/src/lib.rs:7]: Will not make any changes for the moment.

**Pull request merged**